mars/dataframe/expressions/datasource/from_tensor.py

```diff
@@ -0,0 +1,95 @@
+        super(DataFrameFromTensor, self).__init__(_dtypes=dtypes,
+                                                  _gpu=gpu, _sparse=sparse,
+                                                  _object_type=ObjectType.dataframe, **kw)
+
+    @property
+    def dtypes(self):
+        return self._dtypes
+
+    @property
+    def input(self):
+        return self._input
+
+    def _set_inputs(self, inputs):
+        super(DataFrameFromTensor, self)._set_inputs(inputs)
+        self._input = inputs
+
+    def __call__(self, input_tensor):
+        if input_tensor.ndim == 1:
+            # convert to 1-d DataFrame
+            index_value = pd.RangeIndex(start=0, stop=input_tensor.shape[0])
+            columns_value = pd.RangeIndex(start=0, stop=1)
+        elif input_tensor.ndim != 2:
+            raise ValueError('Must pass 1-d or 2-d input')
+        else:
+            # convert to DataFrame
+            index_value = pd.RangeIndex(start=0, stop=input_tensor.shape[0])
+            columns_value = pd.RangeIndex(start=0, stop=input_tensor.shape[1])
+
+        return self.new_dataframe(input_tensor, input_tensor.shape, dtypes=self.dtypes,
+                                  index_value=parse_index(index_value),
+                                  columns_value=parse_index(columns_value))
+
+    @classmethod
+    def tile(cls, op):
+        out_df = op.outputs[0]
+        in_tensor = op.input
+        out_chunks = []
+        nsplits = in_tensor.nsplits
+        if any(any(np.isnan(ns)) for ns in nsplits):
+            raise NotImplementedError('NAN shape is not supported in DataFrame')
+
+        cum_size = [np.cumsum(s) for s in nsplits]
+        for in_chunk in in_tensor.chunks:
+            out_op = op.copy().reset_key()
+            if in_chunk.ndim == 1:
+                i, = in_chunk.index
+                columns_value = pd.RangeIndex(0, 1)
+                index = (in_chunk.index[0], 0)
+            else:
+                i, j = in_chunk.index
+                column_stop = cum_size[1][j]
+                columns_value = pd.RangeIndex(start=column_stop - in_chunk.shape[1], stop=column_stop)
+                index = in_chunk.index
+
+            index_stop = cum_size[0][i]
+            index_value = pd.RangeIndex(start=index_stop - in_chunk.shape[0], stop=index_stop)
+            out_chunk = out_op.new_chunk([in_chunk], shape=in_chunk.shape, index=index,
+                                         index_value=index_value,
+                                         columns_value=columns_value)
+            out_chunks.append(out_chunk)
+
+        new_op = op.copy()
+        return new_op.new_dataframes(None, out_df.shape, dtypes=out_df.dtypes,
+                                     index_value=out_df.index_value,
+                                     columns_value=out_df.columns,
+                                     chunks=out_chunks, nsplits=in_tensor.nsplits)
+
+
+def from_tensor(tensor, gpu=None, sparse=False):
```

Comment:
As `pd.DataFrame` allows user to specify `columns` and `index`, we can allow that too. For now, they cannot be `md.Index`, we can support that later.

Reply:
hi, you mean user just pass **IndexValue** to specifiy columns and index?

Reply:
No, I mean, for now, user can pass pd.Index, np.ndarray or list etc.